### ISSUE TITLE
Newline in assignment causes false negative in `undocumented_unsafe_blocks`

### ISSUE THREAD

**Issue by joshlf** [rust-lang/rust-clippy#11534]
### Summary

When a newline separates the `=` sign and the beginning of an `unsafe` block (as rustfmt sometimes produces), Clippy doesn't recognize a safety comment as decorating the `unsafe` block:

```rust
// SAFETY: ...
let x =
    unsafe { ... };
```

### Lint Name

undocumented_unsafe_blocks

### Reproducer

I tried this code:

```rust
#![deny(clippy::undocumented_unsafe_blocks)]

fn main() {
    // SAFETY: No unsafe code here!
    let _x =
        unsafe { 0 };
    // SAFETY: No unsafe code here!
    let _y = unsafe { 0 };
}
```

I expected to see this happen:

Both `unsafe` blocks should have been recognized as having a safety comment.

Instead, this happened:

`undocumented_unsafe_blocks` fired on `let _x = ...`.

### Version

```text
1.72 (stable)
```

**Comment by y21** [comment]
The expected behavior is behind a configuration option. Setting `accept-comment-above-statement = true` in clippy.toml makes it recognize both safety comments and not lint.
(Also, this would be a FP, wouldn't it? It lints on code where it shouldn't)

**Comment by joshlf** [comment]
> The expected behavior is behind a configuration option. Setting `accept-comment-above-statement = true` in clippy.toml makes it recognize both safety comments and not lint. (Also, this would be a FP, wouldn't it? It lints on code where it shouldn't)

Okay great, thanks!

And yeah, I should have said false positive - I guess I meant that the false negative was failing to notice the safety comment.

**Comment by Alexendoo** [comment]
Is there a reason that isn't on by default?

**Comment by Centri3** [comment]
There's #11170 which changes both to true. Looks like it just needs a `collect-metadata` run

**Comment by Alexendoo** [comment]
Nice, fixed by #11170

**Comment by jstarks** [comment]
This is still broken. It's working for `let` bindings, but not normal assignments.

E.g., this is fine:
```
let x = 
    unsafe { () };
```

This gives a false positive:
```
let x;
x =
    unsafe { () };
```

**Comment by Alexendoo** [comment]
Could you open a new issue for that @jstarks? Looks like it would have a different cause
